### PR TITLE
Improve identifier defintion in the reference

### LINKED ISSUE #28706: Wrong definition of identifiers in Rust Reference
Rust reference says:

>An identifier is any nonempty Unicode2 string of the following form:
>
> * The first character has property XID_start
> * The remaining characters have property XID_continue

The problem with this is that `_` (LOW LINE) is not XID_start (reference: http://www.unicode.org/Public/UCD/latest/ucd/DerivedCoreProperties.txt), but `_ignored` is a valid Rust identifier. Rust lexer explicitly tests for the `_` https://github.com/rust-lang/rust/blob/master/src/libsyntax/parse/lexer/mod.rs#L1399. 

I guess the correct wording would be (keeping in mind that `_` by itself is not a valid identifier): 

```
EIther
   * The first character has property XID_start
   * The remaining characters have property XID_continue
Or
   * The first character is _
   * The second character has property XID_continue
   * The remaining characters have property XID_continue
```

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #28706 

I am pretty sure that this is accurate, bout would love a double check.

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -76,8 +76,12 @@ An identifier is any nonempty Unicode[^non_ascii_idents] string of the following
...
   gated. This is expected to improve soon.
 
-- The first character has property `XID_start`
-- The remaining characters have property `XID_continue`
+Either
+   * The first character has property `XID_start`
+   * The remaining characters have property `XID_continue`
+Or
+   * The first character is `_`
+   * The remaining characters have property `XID_continue`
```
Comment (eefriedman) [inline src/doc/reference.md]: Is `_` by itself an identifier?

Comment (steveklabnik) [inline src/doc/reference.md]: It is not. Should I be more explicit that there must be more than zero 'remaining characters' here?

**Comment by alexcrichton** [comment]
r=me with a clarification of there must be some remaining characters for `_` to start an identifier

**Pull request merged**